docs/design/pulumi-yaml.md

```diff
@@ -0,0 +1,62 @@
+# Integration with Pulumi YAML Programs
+
+## Summary
+
+This design is describes the approach to implementing the ability to execute Pulumi YAML programs from the Operator, by declaring the program as a Kubernetes resource.
+
+## Motivation
+
+Pulumi YAML is intended as an accessible and easy way to write Pulumi programs. 
+YAML is already the standard for creating Kubernetes resources.
+
+Therefore, combining the two means we can provide an easy way to create projects within the Operator, without requiring the use of an external repo for storing a Pulumi project. Projects can be created, edited, and deleted within Kubernetes, making use of kubectl or other automation, as with any other Kubernetes resource.
+
+## Design
+
+Add a new CRD to the Operator, specifying a Pulumi program. This will be [Pulumi YAML](https://www.pulumi.com/docs/reference/yaml/) (as used with the Pulumi CLI) under a 'Program' field, in addition to the usual, compulsory Kubernetes API fields. Additionally add a field to the Stack resource referencing the program, as an alternative to projectRepo (or upcoming Flux fields).
+
+```YAML
+apiVersion: pulumi.com/v1
+kind: Program
+metadata:
+  name: example-program
+program:
+  #Possible requirement for Pulumi project file fields
+  name:
+  runtime:
+  template:
+  ...
+  #Pulumi YAML
+  configuration:
+  resources:
+  variables:
+  outputs:
+```
+
+```YAML
+apiVersion: pulumi.com/v1 #May necessitate a version bump
+kind: Stack
+metadata:
+  name: example-stack
+spec:
+  program: example-program
+  ...
+```
+
+## Alternatives Considered
+
+Two main alternatives were considered: the use of exact Pulumi YAML (rather than 'wrapping' it in a Kubernetes resource), or defining the program within the stack and removing the need for a new resource altogether.
+
+Using Pulumi YAML directly could provide a nicer user experience (and a lower barrier for entry), allowing users to use the same program spec with `pulumi up` and `kubectl apply`. However, this would have required significant work within the Pulumi YAML project itself; requiring it to understand (and appropriately handle/ignore) Kubernetes specific fields. Additionally, adding specially treated fields to the Pulumi YAML specification would likely cause confusion for users. It is also possible that in the future the Pulumi CLI could be adapted to also accept Kubernetes YAML (as defined here), allowing for program resources to be directly run from CLI.
```

Comment:
It's worth explaining why this is a nicer user experience -- what we talked about was the ability to deploy the program as a stack locally, then deploy it to the cluster, without having to have another file that has to be kept up to date.